docs/src/explanations/chisel-enum.md

```diff
@@ -0,0 +1,131 @@
+# ChiselEnum
+
+The ChiselEnum type is an easy way to get away from value encodings and reduce the chance of error when encoding muxes, opcodes, and functional unit operations. As opposed to the Chisel type Enums, ChiselEnums allow for names to be explicitly passed through Bundles like IO. 
+
+## Importing the ChiselEnum module
+
+```scala mdoc
+import chisel3._
+import chisel3.util._
+import chisel3.stage.ChiselStage
+import chisel3.experimental.ChiselEnum
+
+```
+
+## Functionality and Examples
+
+Below we see ChiselEnum being used as mux select for a RISC-V core. While wrapping the object in a package is not required, it is highly recommended as it allows for the type to be used in multiple files more easily. 
+
+```scala mdoc
+// package CPUTypes {
+    object AluMux1Sel extends ChiselEnum {
+        val rs1out, pcout = Value
+        /**
+            "rs1out" -> 0.U,
+            "pcout"  -> 1.U
+        */
+    }
+// }
+```
+
+Here we see a mux using the AluMux1Sel to select between different inputs. 
+
+```scala mdoc
+import AluMux1Sel._
+
+class AluMux1File extends Module {
+    val io = IO(new Bundle {
```

Comment:
i think this is the reason your thing is failing CI, there is a bug that shows up in `mdoc` + naming plugin +` IO( new Bundle {...})`

You can try workaround this by defining your bundle type outside of the IO call:

```
class ALUMuxIO extends Bundle {
...
}
...

val io = IO(new ALUMuxIO)
```

Reply:
Oh that's a pretty nasty issue. I separated the bundle class out and looks like it is passing the ci now. Thanks!